docs/rfcs/2022-08-26-group-connections-recovery.md

```diff
@@ -0,0 +1,43 @@
+# Group connections recovery after asynchronous commands
+
+## Problem
+
+Similar to direct chat connections, group connections can fail to be established on failing IO, for example on bad network. When this happens agent throws error. For direct connections it is viable to propagate the error to UI to indicate failure, so the user can retry. For group connections this becomes unviable as multiple connections are being established automatically on each new joining member (2 for each existing member except host), and the user can't retry.
+
+## Proposal
+
+- Add special versions of commands `createConnection`, `joinConnection` (`allowConnection` as well?), to which agent responds synchronously with connection id and status;
+
+- Add new connection type `NewConnection` / `NoQueueConnection`;
+
+- Add agent responses signalling these commands' asynchronous completion, e.g. `CREATE_SUCCESS`, `JOIN_SUCCESS` (?);
+
+- Return connection status on subscriptions;
+
+In chat:
+
+- Event-driven processing - for group connections use commands with synchronous responses, save intermediate connection state, process success responses;
```

Comment:
```suggestion
- Event-driven processing - for group connections use commands with asynchronous responses, save intermediate connection state, process success responses;
```

Reply:
meant immediate synchronous response, e.g. conn id, and the "success response" is asynchronous